xbmc/windowing/wayland/WinSystemWayland.h

```diff
@@ -117,6 +117,8 @@ class CWinSystemWayland : public CWinSystemBase,
   virtual void SetContextSize(CSizeInt size) = 0;
   virtual IShellSurface* CreateShellSurface(const std::string& name);
 
+  std::unique_ptr<IShellSurface> m_shellSurface;
```

Comment:
data members should not be `protected` unless necessary, a protected getter function for the `IShellSurface*` would be better. this allows subclasses to use the shell surface, but not to mess with the unique_ptr.

Reply:
Added a getter function returning `IShellSurface*`. I was wondering though if returning a `unique_ptr<IShellSurface>&` would be better similar to the other getter functions

Reply:
that would allow callers to mess with the unique_ptr itself (e.g. reset it, even by accident), so I think a plain pointer is better

Reply:
True, though that can happen with the raw pointer too. We can return either `const std::unique_ptr<IShellSurface>&` or `const IShellSurface*` instead.

Reply:
not really, maybe you misunderstood? I was talking about the unique_ptr instance itself, not what it points to.

If you return a `const std::unique_ptr<T>&`, it's basically the same as returning `T*` from what you can do with it, just more complicated and limiting the internal implementation to a unique_ptr :) If you return a ` const IShellSurface*`, you will probably not be able to call the functions you need to any more, since the object itself will be `const`.